dev_guide/dev_tutorials/binary_builds.adoc

```diff
@@ -0,0 +1,307 @@
+  pushed anywhere.
+
+- Building private code - New builds can be started from scratch as binary builds. The source can then
+  be uploaded directly from your local workstation to OpenShift without having to check it in to an SCM. 
+
+- Building images with artifacts from other sources - With Jenkins pipelines, binary builds are useful
+  to combine artifacts built with tools such as maven, C compiler, etc. and runtime images that make use
+  of those builds.
+
+[[binary-builds-limitations]]
+=== Limitations
+
+- Binary builds are not repeatable
+  Because binary builds rely on the user uploading artifacts at build start, OpenShift cannot repeat
+  the same build without the user repeating the same upload every time.
+
+- Binary builds cannot be triggered automatically
+  They can only be started manually when the user uploads the required binary artifacts. 
+
+[NOTE]
+====
+Builds that are started as binary builds may also have a configured source URL. If that's the case, triggers
+will successfully launch the build but source will come from the configured source URL and not from 
+what was supplied by the user the last time the build ran.
+====
+
+[[binary-builds-tutorial-overview]]
+== Tutorials Overview
+
+The following tutorials assume that you have an OpenShift cluster available and that you have a 
+project where you can create artifacts. It requires that you have both `git` and `oc` available locally.
+
+
+[[binary-builds-local-code-changes]]
+=== Tutorial: Building local code changes
+
+. Create a new application based on an existing source repository and create a route for it.
++
+----
+oc new-app https://github.com/openshift/ruby-hello-world.git
+oc expose svc/ruby-hello-world
+----
+
+. Wait for the initial build to complete and view the application's page by navigating to the route's host.
+  You should get a welcome page.
++
+----
+oc get route ruby-hello-world
+----
+
+. Clone the repository locally
++ 
+----
+git clone https://github.com/openshift/ruby-hello-world.git
+cd ruby-hello-world
+----
+
+. Make a change to the application's view.
+  Using your favorite editor, edit `views/main.rb`:
+  +
+  Change the `<body>` tag to `<body style="background-color:blue">`
+
+. Start a new build with your locally-modified source. From the repository's
+  local directory, run:
+  +
+  ----
+  oc start-build ruby-hello-world --from-dir="." --follow
+  ----
+
+Once your build has completed and the application has redeployed, navigating to the application's route host should 
+result in a page with a blue background.
+
+You can keep making changes locally and building your code with `oc start-build --from-dir`.
+
+You can also create a branch of the code, commit your changes locally, and use the repository's HEAD as the source
+for your build:
+ 
+----
+git checkout -b my_branch
+git add .
+git commit -m "My changes"
+oc start-build ruby-hello-world --from-repo="." --follow
```

Comment:
give a little more explanation about what this does.

Reply:
explanation added below